chia/consensus/blockchain.py

```diff
@@ -268,22 +269,26 @@ async def run_single_block(self, block: FullBlock, fork_info: ForkInfo) -> None:
         assert fork_info.peak_height == block.height - 1
         assert block.height == 0 or fork_info.peak_hash == block.prev_header_hash
 
-        additions: list[tuple[Coin, Optional[bytes]]] = []
-        removals: list[Coin] = []
+        conds = None
         if block.transactions_generator is not None:
             block_generator: Optional[BlockGenerator] = await get_block_generator(self.lookup_block_generators, block)
             assert block_generator is not None
             assert block.transactions_info is not None
             assert block.foliage_transaction_block is not None
-            flags = get_flags_for_height_and_constants(block.height, self.constants)
-            additions, removals = additions_and_removals(
-                bytes(block.transactions_generator),
-                block_generator.generator_refs,
-                flags,
-                self.constants,
+            flags = get_flags_for_height_and_constants(block.height, self.constants) | DONT_VALIDATE_SIGNATURE
+            err, conds = run_block_generator2(
```

Comment:
can you explain using the (slower) `run_block_generator2()` instead of the (faster) `additions_and_removals()`?

Reply:
also, I would expect this to break for pre-hardfork blocks, when we use `run_block_generator()`